e2e-test/e2e/chaos/httpchaos/http_tls.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package httpchaos
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/e2e-test/e2e/util"
+)
+
+//go:embed keys
+var content embed.FS
+
+type TLSServerKeys struct {
+	Cert []byte `json:"cert"`
+	Key  []byte `json:"key"`
+}
+
+func setupHTTPS(cli *http.Client, serverIP string) (TLSServerKeys, []byte) {
+	c, err := content.ReadDir("keys")
+	framework.ExpectNoError(err, "read key dir error")
+	var key []byte
+	var ca []byte
+	for _, f := range c {
+		if f.IsDir() {
+			continue
+		}
+		b, err := content.ReadFile("keys/" + f.Name())
+		framework.ExpectNoError(err, "read key file error")
+		switch f.Name() {
+		case "server.key":
+			key = b
+		case "ca.crt":
+			ca = b
+		}
+		err = os.WriteFile(f.Name(), b, 0644)
+		framework.ExpectNoError(err, "write key file error")
+	}
+
+	f, err := os.OpenFile("server.ext", os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		framework.ExpectNoError(err, "open server.ext file error")
+	}
+	if _, err = f.WriteString(fmt.Sprint("IP.1 = " + serverIP)); err != nil {
+		framework.ExpectNoError(err, "write server.ext file error")
+	}
+	err = f.Close()
+	framework.ExpectNoError(err, "close server.ext file error")
+
+	cmdStr := "openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -days 3650 -sha256 -extfile server.ext"
+	cmd := exec.Command("bash", "-c", cmdStr)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		framework.ExpectNoError(err, "run openssl cmd error: "+string(output))
+	}
```

Comment:
it seems that server.crt already existed in directory `keys` 🤔

Reply:
I will keep them , and then someone can just create a serect directly with it.